File: app/src/main/java/ru/samtakoy/core/business/QPacksExporter.java
package ru.samtakoy.core.business;

import io.reactivex.Completable;
import ru.samtakoy.core.model.QPack;

public interface QPacksExporter {

    //void exportThemeTree(QPack qPack);
    boolean exportQPackToEmail(QPack qPack);
    boolean exportQPack(QPack qPack);

    Completable exportAllToEmail();
    Completable exportAllToFolder(String exportDirPath);

}

File: module/domain/export/api/src/commonMain/kotlin/ru/samtakoy/domain/exportcards/QPacksExporter.kt
package ru.samtakoy.domain.exportcards

import ru.samtakoy.domain.qpack.QPack

interface QPacksExporter {
    // @WorkerThread
    suspend fun exportQPackToEmail(qPackId: Long)

    // @WorkerThread
    suspend fun exportQPack(qPack: QPack)

    // @WorkerThread
    suspend fun exportAllToEmail()

    // @WorkerThread
    suspend fun exportAllToFolder(exportDirPath: String)
}
